### PR TITLE
Add InputSetupWizard component and context

### PR DESCRIPTION
**Pull request by grotlue**
Closes https://github.com/Graylog2/graylog2-server/issues/20531

## Description
/nocl unreleased feature and code cleanup

In order to prepare for the InputSetupWizard this PR
- adds a scaffold component for the wizard, which can retain wizard and step-related state in a context. This component can potentially be reused for the upcoming Data Source wizard by adding dynamic typing
- migrates existing input components to typescript and functional components. I did only migrate existing code, with only a bit of cleanup as we are planning to rebuild the input list soon.

## How Has This Been Tested?

We need to test:
- Input creation (custom inputs like AWS and inputs using the configuration form like Random HTTP)
- Input edit form
- All actions on an input list item

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/inputs/CreateInputControl.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import { TELEMETRY_EVENT_TYPE } from 'logic/telemetry/Constants';
+import useSendTelemetry from 'logic/telemetry/useSendTelemetry';
+import useLocation from 'routing/useLocation';
+import { Col, Row, Button } from 'components/bootstrap';
+import { ExternalLinkButton, Select } from 'components/common';
+import { InputForm } from 'components/inputs';
+import type { Input } from 'components/messageloaders/Types';
+
+const NewInputRow = styled(Row)`
+  margin-bottom: 8px;
```
Comment (gally47) [inline graylog2-web-interface/src/components/inputs/CreateInputControl.tsx:37]: Can we use theme spacing here?

File: graylog2-web-interface/src/components/inputs/CreateInputControl.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    });
+  };
+
+  const CustomInputsConfiguration = customInputConfiguration ? customInputConfiguration.component : null;
+
+  return (
+    <NewInputRow className="content">
+      <Col md={12}>
+        <StyledForm className="form-inline" onSubmit={handleInputTypeSubmit}>
+          <div className="form-group" style={{ width: 300 }}>
```
Comment (gally47) [inline graylog2-web-interface/src/components/inputs/CreateInputControl.tsx:130]: Can we use styled components here?

File: graylog2-web-interface/src/contexts/InputSetupWizardContext.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  TEST_INPUT: 'TEST_INPUT',
+  SETUP_ROUTING: 'SETUP_ROUTING',
+  CREATE_STREAM: 'CREATE_STREAM',
+  COMPLETE: 'COMPLETE',
+} as const;
+
+export type InputSetupWizardStep = typeof INPUT_WIZARD_STEPS[keyof typeof INPUT_WIZARD_STEPS]
+
+export type StepsData = {
+  [key in InputSetupWizardStep]?: any
```
Comment (gally47) [inline graylog2-web-interface/src/contexts/InputSetupWizardContext.tsx:34]: can we type this instead of any?

**Review by gally47 — CHANGES_REQUESTED** [review]
Tested it. Looking good, found few nitpicks below:

File: graylog2-web-interface/src/contexts/InputSetupWizardContext.tsx (hunk 3 above)
Comment (grotlue) [reply graylog2-web-interface/src/contexts/InputSetupWizardContext.tsx:34]: Yes, but first we need the steps, for now this is just a skeleton.

File: graylog2-web-interface/src/components/inputs/CreateInputControl.tsx (hunk 1 above)
Comment (grotlue) [reply graylog2-web-interface/src/components/inputs/CreateInputControl.tsx:37]: Actually I realised this is obsolete as this styling is overridden by the content class

**Review by gally47 — APPROVED** [review]
LGTM.

**Pull request merged**